File: 1222/1222.c
// bee 1222 - Concurso de Contos
// Alonso Martins
// 02/05/2024
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main()
{
    int N, L, C, i;
    int caracteres, linhas, paginas, len;
    char palavra[80];

    while (scanf("%d %d %d%*c", &N, &L, &C) != EOF)
    {
        caracteres = 0; linhas = paginas = 1;


        for (i = 0; i < N; i ++)
        {
            scanf("%s", palavra); len = strlen(palavra);
            caracteres += len + 1;

            if (caracteres - 1 > C)
            {
                caracteres = len + 1;
                linhas ++;
            }
            if (linhas > L)
            {
                linhas = 1;
                paginas ++;
            }
        }

        printf("%d\n", paginas);
    }

    return 0;
}

File: 1222/1222.py
# bee 1222 - Concurso de Contos
# Alonso Martins
# 02/05/2024
while True:
    try:
        N, L, C = map(int, input().split())
        caracteres, linhas, paginas = 0, 1, 1
        conto = input().split()
        
        for palavra in conto:
            size = len(palavra)
            caracteres += size + 1
            
            if caracteres - 1 > C:
                caracteres = size + 1
                linhas += 1

            if linhas > L:
                linhas = 1
                paginas += 1
                
        print(paginas)

    except EOFError:
        break
